### PR TITLE
i18n support in devconsole helm components

### PR DESCRIPTION
**Pull request by vikram-raj**
fixes: https://issues.redhat.com/browse/ODC-5084

### CODE REVIEW

File: frontend/packages/dev-console/src/components/helm/form/HelmChartVersionDropdown.tsx [hunk 1]
```diff
@@ -48,22 +50,23 @@ const HelmChartVersionDropdown: React.FunctionComponent<HelmChartVersionDropdown
...
     confirmModal({
-      title: 'Change Chart Version?',
+      title: t('devconsole~Change Chart Version?'),
       message: (
         <>
           <p>
-            Are you sure you want to change the chart version from <strong>{currentVersion}</strong>{' '}
-            to <strong>{newVersion}</strong>?{' '}
+            {t('devconsole~Are you sure you want to change the chart version from')}{' '}
+            <strong>{currentVersion}</strong> {t('devconsole~to')} <strong>{newVersion}</strong>?{' '}
```
Comment (christianvogt) [inline frontend/packages/dev-console/src/components/helm/form/HelmChartVersionDropdown.tsx:58]: This should be translated as a single string.

File: frontend/packages/dev-console/src/components/helm/form/HelmChartVersionDropdown.tsx [hunk 2]
```diff
@@ -48,22 +50,23 @@ const HelmChartVersionDropdown: React.FunctionComponent<HelmChartVersionDropdown
...
           </p>
           <p>
-            <InfoCircleIcon color="var(--pf-global--info-color--100)" /> All data entered for
-            version <strong>{currentVersion}</strong> will be reset.
+            <InfoCircleIcon color="var(--pf-global--info-color--100)" />{' '}
+            {t('devconsole~All data entered for version')} <strong>{currentVersion}</strong>{' '}
+            {t('devconsole~will be reset')}.
```
Comment (christianvogt) [inline frontend/packages/dev-console/src/components/helm/form/HelmChartVersionDropdown.tsx:63]: This should be translated as a single string.

File: frontend/packages/dev-console/src/components/helm/form/HelmReleaseRollbackForm.tsx [hunk 3]
```diff
@@ -29,12 +30,14 @@ const HelmReleaseRollbackForm: React.FC<Props> = ({
...
+  const { t } = useTranslation();
   const { type: helmAction, title } = helmActionConfig;
 
   const formHelpText = (
     <>
-      Select the version to rollback <strong style={{ color: '#000' }}>{releaseName}</strong> to,
-      from the table below:
+      {t('devconsole~Select the version to rollback')}{' '}
+      <strong style={{ color: '#000' }}>{releaseName}</strong>{' '}
+      {t('devconsole~to, from the table below:')}
```
Comment (christianvogt) [inline frontend/packages/dev-console/src/components/helm/form/HelmReleaseRollbackForm.tsx:40]: Translate as a single string.

File: frontend/packages/dev-console/src/components/helm/form/HelmInstallUpgradeForm.tsx [hunk 4]
```diff
@@ -65,7 +67,7 @@ const HelmInstallUpgradeForm: React.FC<FormikProps<FormikValues> & HelmInstallUp
 
   const readmeText = chartReadme && (
     <>
-      For more information on the chart, refer to this{' '}
+      {t('devconsole~For more information on the chart, refer to this')}{' '}
```
Comment (christianvogt) [inline frontend/packages/dev-console/src/components/helm/form/HelmInstallUpgradeForm.tsx:70]: Translate as a single string.

File: frontend/packages/dev-console/src/components/helm/details/history/HelmReleaseHistoryRow.tsx [hunk 5]
```diff
@@ -28,7 +29,7 @@ const confirmModalRollbackHelmRelease = (
   const executeFn = () => coFetchJSON.patch('/api/helm/release', payload);
 
   return {
-    label: `Rollback to Revision ${revision}`,
+    label: `devconsole~Rollback to Revision ${revision}`,
```
Comment (debsmita1) [inline frontend/packages/dev-console/src/components/helm/details/history/HelmReleaseHistoryRow.tsx:32]:
```suggestion
    label: t('devconsole~Rollback to Revision {{revision}}',{revision})
```

**Comment by debsmita1** [comment]
localize strings in `concatVersions` in `helm-utils.ts`
`HelmReleaseRollbackForm.tsx` -> FormGroup label, aria-label & resetLabel
`HelmReleaseResourcesRow.tsx` -> Link title
`modify-helm-release.ts`

**Review by debsmita1 — CHANGES_REQUESTED** [review]

File: frontend/packages/dev-console/src/components/helm/form/HelmInstallUpgradeForm.tsx [hunk 6]
```diff
@@ -140,7 +142,7 @@ const HelmInstallUpgradeForm: React.FC<FormikProps<FormikValues> & HelmInstallUp
         isSubmitting={status?.isSubmitting || isSubmitting}
         submitLabel={helmAction}
```
Comment (jerolimov) [inline frontend/packages/dev-console/src/components/helm/form/HelmInstallUpgradeForm.tsx:143]: We need to translate the `submitLabel` as well. `helmAction` is an enum which we could not translate directly, so we need a mapping here from enum to label, or?

File: frontend/packages/dev-console/src/components/helm/helm-utils.ts [hunk 7]
```diff
@@ -123,25 +124,29 @@ export const getChartEntriesByName = (
...
 ): string => {
   let title = chartVersion.split('--')[0];
   if (appVersion) {
-    title += ` / App Version ${appVersion}`;
+    title += t('devconsole~ / App Version {{appVersion}}', { appVersion });
   }
   if (chartRepoName) {
-    title += ` (Provided by ${toTitleCase(chartRepoName)})`;
+    title += t('devconsole~ (Provided by {{toTitleCase}})', {
+      toTitleCase: toTitleCase(chartRepoName),
```
Comment (jerolimov) [inline frontend/packages/dev-console/src/components/helm/helm-utils.ts:136]: Please give this localization variable a cleaner name, `chartRepoName` or `repoName`.

Needs to be changed on some tests as well

**Comment by jerolimov** [comment]
Test fails because of    ✖ Update Annotation from empty value to value (1 failure)

/retest

File: frontend/packages/dev-console/src/components/helm/form/HelmInstallUpgradeForm.tsx (hunk 6 above)
Comment (vikram-raj) [reply frontend/packages/dev-console/src/components/helm/form/HelmInstallUpgradeForm.tsx:143]: We can no translate this enum as it also use to show the form type of `Install`, `Upgrade` or `Rollback`.

Comment (jerolimov) [reply frontend/packages/dev-console/src/components/helm/form/HelmInstallUpgradeForm.tsx:143]: What's about 

```suggestion
        submitLabel={resetLabel={t(`devconsole~${helmAction}`)}}
```

?

Not perfect but should work. Don't know if we have some other areas where we need to translate enums. But if I'm right the good thing is that if we add an enum the translation shows as label (helmaction) as fallback. In our case we need to add all three labels also if they doesn't existing 1:1 in sourcecode. (I mean, there is no reverse lookup like t(`devconsole~Install`))

Comment (jerolimov) [reply frontend/packages/dev-console/src/components/helm/form/HelmInstallUpgradeForm.tsx:143]: Same in `HelmReleaseRollbackForm.tsx`

**Review by jerolimov — APPROVED** [review]
Tested this locally and looks good. @debsmita1 are you also fine with your requested changes?

/lgtm
/retest

**Comment by invincibleJai** [comment]
/approve

Verified the changes

**Comment by christianvogt** [comment]
@vikram-raj front end tests failing due to duplicate key

**Review by jerolimov — APPROVED** [review]
/lgtm
/retest

**Pull request merged**